core/templates/list.h

```diff
@@ -135,6 +135,83 @@ class List {
 		_FORCE_INLINE_ Element() {}
 	};
 
+	typedef T ValueType;
+
+	struct Iterator {
+		_FORCE_INLINE_ T &operator*() {
+			return E->get();
+		}
+		_FORCE_INLINE_ T *operator->() { return &E->get(); }
```

Comment:
I think `operator*` and `operator->` of non-const iterators could also be marked `const`, they does not modify the iterator itself.

```c++
const Iterator iter = mutable_container.begin();
T value = *iter;  // should be possible
```